File: Java Implementation/src/P12.java
//Highly divisible triangular number
public class P12 {

  public static void main(String[] args) {
    //if triangleNum divisible by n, count++, n++, count <= 500
    int count = 0;
    for (int i = 1; count < 500; i++) {
      count = 0;
      //System.out.println("Sequence " + i + " has triangle number " + triangleNum(i));
      for (int j = 1; j <= triangleNum(i); j++) {
        if (triangleNum(i) % j == 0) {
          count++;
        }
      }
      System.out.println("Triangle number " + triangleNum(i) + " of sequence " + i + " has " + count + " divisors");
    }

  }

  public static int triangleNum(int n) {
    int num = 0;
    for (int i = 1; i <= n; i++) {
      num += i;
    }
    return num;
  }

}

File: Swift Implementation/P12.swift
//
//  P12.swift
//  
//
//  Created by Xin Jet Chew on 17/01/2017.
//
//

import Foundation

var highestDivisorNum = 0
var index = 0
var triNum = 0

repeat {
    index += 1
    var numOfDivisors = 0
    triNum += index
    if triNum < 10 {
        numOfDivisors += 1
        for divisor in 0 ... triNum / 2  {
            if divisor > 0 {
                if triNum % divisor == 0 {
                    numOfDivisors += 1
                }
            }
        }
    }
    else {
        for divisor in 0 ..< Int(sqrt(Double(triNum))) {
            if divisor > 0 {
                if triNum % divisor == 0 {
                    numOfDivisors += 2
                }
            }
        }
    }
    if numOfDivisors > highestDivisorNum {
        highestDivisorNum = numOfDivisors
        print("Triangular number \(triNum) has \(highestDivisorNum) number of divisors")
    }
}
while(highestDivisorNum < 500)
